collections/_development_standards/patterns.md

```diff
@@ -169,3 +169,49 @@ Another positive side-effects of using CQRS is that there are naturally less mer
 
 * [CQRS - Microsoft](https://docs.microsoft.com/en-us/azure/architecture/patterns/cqrs)
 * [CQRS - Martin Fowler](https://martinfowler.com/bliki/CQRS.html)
+
+
+### Managed Identities (MI)
+
+Using a managed identity allows an application to authenticate to any service that supports Azure AD authentication without having credentials.
+
+* Secretless configuration
+* Acquiring a bearer token for an API behund Azure AD authentication
+* Connecting to a SQL database
+
+The ```Microsoft.Azure.Services.AppAuthentication``` library manages authentication automatically
+
+#### Example - Authenticating with another app service, eg internal API
+
+    public class ApiClient
+    {
+        private readonly HttpClient _httpClient;
+
+        public ApiClient(IHttpClientFactory httpClientFactory) {
+            _httpClient = httpClientFactory.CreateClient();
+        }
+
+        public async Task<string> GetAccessTokenAsync(string identifier)
+        {
+            var azureServiceTokenProvider = new AzureServiceTokenProvider();
+            var accessToken = await azureServiceTokenProvider.GetAccessTokenAsync(identifier);
+            
+            return accessToken;
+        }
+
+        # Acquire a bearer token
+        var identifier = "https://tenant.onmicrosoft.com/das-env-api-as-ar" # IdentifierUri of Azure AD App registration
+        var accessToken = await GetAccessTokenAsync(identifier);
+
+        # Add the token as a header
+        _httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", accessToken);
+
+        # Make an API call
+        var response = await _httpClient.GetAsync("https://azurewebsites.net/resource/resourceid");
+    }
+
+#### References
+
+* [MI Overview - Microsoft](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview)
```

Comment:
I would add the root URL of the MS docs as a reference as it gives a bit more info and has further links to other useful resources for MI:
https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/

Reply:
Added link